parsel/unified.py

```diff
@@ -66,26 +53,28 @@ class Selector(object_ref):
     }
     _lxml_smart_strings = False
 
-    def __init__(self, response=None, text=None, type=None, namespaces=None,
-                 _root=None, _expr=None):
-        self.type = st = _st(response, type or self._default_type)
+    def __init__(self, text=None, type=None, namespaces=None, root=None, _expr=None):
+        self.type = st = _st(type or self._default_type)
         self._parser = _ctgroup[st]['_parser']
         self._csstranslator = _ctgroup[st]['_csstranslator']
         self._tostring_method = _ctgroup[st]['_tostring_method']
 
         if text is not None:
-            response = _response_from_text(text, st)
-
-        if response is not None:
-            _root = LxmlDocument(response, self._parser)
+            root = self._get_root(text)
+        elif root is None:
+            raise ValueError("Selector needs either text or root argument")
 
-        self.response = response
         self.namespaces = dict(self._default_namespaces)
         if namespaces is not None:
             self.namespaces.update(namespaces)
-        self._root = _root
+        self._root = root
         self._expr = _expr
 
+    def _get_root(self, text):
+        body = text.strip().encode('utf8') or u'<html/>'
```

Comment:
This doesn't look correct: if text is empty body becomes unicode (`u'<html/>'`), but if text is not empty body becomes a bytestring (`foo.encode('..')`).

AFAIK we require unicode here to unify data to utf8 before passing it to lxml. LXML can work on bytes (it'll try to detect the encoding), but we can do it better because there are other signs available like HTTP headers.